syntax = "proto3";
package net.phoneyou.covid19.bb.descriptor;

import "amster/gbbinfra/DescriptorOptions.proto";
import "net/phoneyou/covid19/bb/Common.proto";

message Statistics{
    option (amster.gbbinfra.descriptor_type) = 2001;
    option (amster.gbbinfra.singleton) = true ;

    int32 injected_cnt = 1; // 被感染数
    int32 removed_cnt = 2; // 接触感染
    int32 susceptible_cnt = 3; // 易感染数
}

// 传染病模型参数
message LemologyModel{
    option (amster.gbbinfra.descriptor_type) = 2002;
    option (amster.gbbinfra.singleton) = true ;

    DurationOptions duration = 1;
    RateOptions rate = 2;
    bool sis_model = 3; // 采用SIS模式，没有Removed状态
    uint32 contact_degree = 4;
    bool inited = 5; // TODO shuold not be here
    bool sync_duration_time = 6;
}

message Dimensions{
    option (amster.gbbinfra.descriptor_type) = 2003;
    option (amster.gbbinfra.singleton) = true ;

    uint32 cell_x = 1;
    uint32 cell_y = 2;
}
